zircon.core/common/src/main/kotlin/org/hexworks/zircon/internal/component/impl/DefaultSlider.kt

```diff
@@ -0,0 +1,254 @@
+
+    override fun acceptsFocus() = isDisabled.not()
+
+    override fun focusGiven() = whenEnabled {
+        LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was given focus.")
+        componentStyleSet.applyFocusedStyle()
+        render()
+    }
+
+    override fun focusTaken() = whenEnabled {
+        LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) lost focus.")
+        componentStyleSet.reset()
+        render()
+    }
+
+    override fun mouseEntered(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse entered.")
+            componentStyleSet.applyMouseOverStyle()
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mouseExited(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse exited.")
+            componentStyleSet.reset()
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mousePressed(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse pressed.")
+            componentStyleSet.applyActiveStyle()
+            setValueToClosestOfStep(getMousePosition(event))
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mouseReleased(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse released.")
+            componentStyleSet.applyMouseOverStyle()
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mouseDragged(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse pressed.")
+            componentStyleSet.applyActiveStyle()
+            setValueToClosestOfStep(getMousePosition(event))
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun keyPressed(event: KeyboardEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            when (event.code) {
+                KeyCode.RIGHT -> {
+                    incrementCurrentValue()
+                    Processed
+                }
+                KeyCode.LEFT -> {
+                    decrementCurrentValue()
+                    Processed
+                }
+                KeyCode.UP -> {
+                    addToCurrentValue(valuePerStep.roundToInt())
```

Comment:
What's the conceptual difference between `increment`/`decrement` and `add`/`subtract`? One always uses +-1 and the other uses `valuePerStep`?

Reply:
increment/decrement is by one, add/substract is by an arbitrary value (here we're adding a whole step but it could be anything). Could use some change. I find that increment/decrement are a nice way to interface a button